unit_tests/test_aggregator.py

```diff
@@ -0,0 +1,411 @@
+
+TEST_AREA_NAME = "TestArea"
+TEST_AREA_MAPPING = {TEST_AREA_NAME: [str(uuid.uuid4())]}
+
+TEST_BATCH_COMMAND_DICT = {"some_device_uuid": [{"command_dict0"},
+                                                {"command_dict1"}],
+                           "some_other_device_uuid": [{"command_dict2"},
+                                                      {"command_dict3"}],
+                           }
+
+TEST_BATCH_COMMAND_RESPONSE = {"responses": {"asset_id_1": ["response1", "response2"],
+                                             "asset_id_2": ["response3", "response4"]}}
+
+TEST_CONFIGURATION_PREFIX = "/configuration_prefix/"
+
+TEST_AGGREGATOR_PREFIX = "/aggregator_prefix/"
+
+TEST_JWT_KEY_FROM_SERVER = str(uuid.uuid4())
+TEST_TRANSACTION_ID = str(uuid.uuid4())
+
+
+@pytest.fixture(autouse=True, name="mock_connections")
+def fixture_mock_connections(mocker):
+    """Mock methods and functions which establish external connections."""
+    mocker.patch("gsy_e_sdk.clients.rest_asset_client.retrieve_jwt_key_from_server",
+                 return_value=TEST_JWT_KEY_FROM_SERVER)
+    mocker.patch("gsy_framework.client_connections.utils.RestCommunicationMixin"
+                 "._create_jwt_refresh_timer")
+    mocker.patch("gsy_e_sdk.aggregator.Aggregator._connect_to_simulation",
+                 autospec=True)
+
+
+@pytest.fixture(name="mock_prefixes")
+def fixture_mock_prefixes(mocker):
+    """Mock external prefix calculation functions."""
+    mocker.patch("gsy_e_sdk.clients.rest_asset_client.get_aggregator_prefix",
+                 return_value=TEST_AGGREGATOR_PREFIX, autospec=True)
+    mocker.patch("gsy_e_sdk.clients.rest_asset_client.get_configuration_prefix",
+                 return_value=TEST_CONFIGURATION_PREFIX)
+
+
+@pytest.fixture(name="mock_start_websocket_funcs")
+def fixture_mock_start_websocket_funcs(mocker):
+    """Mock external classes used in Aggregator.start_websocket_connections."""
+    mocker.patch("gsy_e_sdk.aggregator.AggregatorWebsocketMessageReceiver")
+    mocker.patch("gsy_e_sdk.aggregator.WebsocketThread")
+    mocker.patch("gsy_e_sdk.aggregator.ThreadPoolExecutor")
+
+
+@pytest.fixture(name="mock_execute_batch_command_methods")
+def fixture_mock_execute_batch_command_methods(mocker):
+    """Mock methods and external functions and other methods used in
+    execute_batch_command."""
+    mocker.patch("gsy_e_sdk.aggregator.ClientCommandBuffer.buffer_length",
+                 new_callable=PropertyMock, return_value=3)
+    mocker.patch("gsy_e_sdk.aggregator.ClientCommandBuffer.execute_batch",
+                 return_value=TEST_BATCH_COMMAND_DICT)
+    mocker.patch("gsy_e_sdk.aggregator.ClientCommandBuffer.clear")
+    mocker.patch("gsy_e_sdk.aggregator.Aggregator._post_request",
+                 return_value=[TEST_TRANSACTION_ID, True])
+    mocker.patch("gsy_e_sdk.aggregator.Aggregator."
+                 "_all_uuids_in_selected_device_uuid_list",
+                 return_value=True)
```

Comment:
I lost a bit track of this PR. Could you please explain why we're patching this private method? Was this something that was waiting for refactoring?

Reply:
Addressed on [0acc18e](https://github.com/gridsingularity/gsy-e-sdk/pull/194/commits/0acc18eb839cc01a84592e34ddd84dcff7f9b4e0)